src/spatial/models/heatmapModel.py

```
@@ -618,12 +635,20 @@
         try:
-            joblib.dump(model, model_path)
-            self.logger.info(f"Saved model for '{city_name}' to {model_path}")
+        # Save locally first
+            joblib.dump(model, local_path)
+            self.logger.info(f"Saved model locally for '{city_name}' to {local_path}")
+         # If running in production (bucket available), upload to GCS
+            if self.SPATIAL_PROJECT_BUCKET:
+                gcs_path = f"{self.SPATIAL_PROJECT_BUCKET}/{model_filename}"
+                upload_to_gcs(local_path, gcs_path)
+                self.logger.info(f"Uploaded model for '{city_name}' to {gcs_path}")
         except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)